GenumKit/Parsers/FontsFileParser.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  FontsFileParser.swift
+//  Pods
+//
+//  Created by Derek Ostrander on 3/7/16.
+//
+//
+
+import Foundation
+import AppKit.NSFont
+
+typealias Font = (name: String, variant: String)
+
+extension CFString {
+    var alphanumericString: String {
+         return (self as String).componentsSeparatedByCharactersInSet(NSCharacterSet.alphanumericCharacterSet().invertedSet).joinWithSeparator("")
+    }
+}
+
+extension CTFont {
+    static func loadFont(path: String) -> Font? {
+        let inData = NSData(contentsOfFile: path)
+        var error: Unmanaged<CFError>?
+        guard
+            let provider: CGDataProvider = CGDataProviderCreateWithCFData(inData),
```

Comment:
Ok, I came up with a way better way to load fonts:

* It support more type of fonts (I tested by iterating over all the fonts in `/Library/Fonts` and for some of them, creating the intermediate `CGDataProvider` or the `CGFont` fail, whereas with this new code it works all the time)
* It extracts the font variant using the official name, instead of trying to guess it from the full name

Here's the code I used (tested in playground):

```swift
extension CTFont {
  static func loadFont(fileURL: NSURL) -> Font? {
    let descs = CTFontManagerCreateFontDescriptorsFromURL(fileURL) as NSArray?
    guard let desc = descs?.firstObject else { return nil }
    let font = CTFontCreateWithFontDescriptorAndOptions(desc as! CTFontDescriptor, 0.0, nil, [.PreventAutoActivation])
    guard let family = CTFontCopyAttribute(font, kCTFontFamilyNameAttribute) as? String else { return nil }
    guard let variant = CTFontCopyAttribute(font, kCTFontStyleNameAttribute) as? String else { return nil }
    return Font(name: family, variant: variant)
  }
}
```

Then no need to build a data provider, build an intermediate `CGFont` register the `CTFont` or whatnot, and you can now get rid of the `alphanumericString`, `private var font: Font?` helpers and stop guessing the font variant!